src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3112,51 +3123,61 @@ void AwtWindow::_SetFocusableWindow(void *param)
     delete sfws;
 }
 
-void AwtWindow::_ModalDisable(void *param)
-{
-    JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
+void AwtWindow::_ModalDisable(void *param) {
+    JNIEnv *env = (JNIEnv *) JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    ModalDisableStruct *mds = (ModalDisableStruct *)param;
+    ModalDisableStruct *mds = static_cast<ModalDisableStruct *>(param);
     jobject self = mds->window;
     HWND blockerHWnd = (HWND)mds->blockerHWnd;
 
     AwtWindow *window = NULL;
     HWND windowHWnd = 0;
 
-    JNI_CHECK_NULL_GOTO(self, "peer", ret);
     PDATA pData = JNI_GET_PDATA(self);
-    if (pData == NULL) {
-        env->DeleteGlobalRef(self);
+    if (self == NULL) {
```

Comment:
Surely line 3136 above should have been deleted ? It is replaced by line 3143 below.
And then you can also directly set window, pData isn't needed, as discussed in similar cases above

Reply:
Yes. 

@TheShermanTanker This is in fact a serious bug. You try to do `pData = JNI_GET_PDATA(self)` before the null check of self, and this will crash if self is null.